### PR TITLE
feat(metrics): Support min/max params on metrics samples

### PR DESCRIPTION
**Pull request by Zylphrex**
When the user selects a focus area, we want to narrow the search to just values within the focus area. This will be done by adjusting the data params and adding a new min/max param to specify the bounds. This PR adds support for this.

### CODE REVIEW

File: src/sentry/sentry_metrics/querying/samples_list.py [hunk 1]
```diff
@@ -287,13 +301,23 @@ def execute(self, offset, limit):
...
     def get_span_keys(self, offset: int, limit: int) -> list[tuple[str, str, str]]:
+        column = self.mri_to_column(self.mri)
+
+        query_parts = []
+        if self.query:
+            query_parts.append(self.query)
+        if self.min is not None:
+            query_parts.append(f"{column}:>={self.min}")
+        if self.max is not None:
+            query_parts.append(f"{column}:<={self.max}")
```
Comment (wmak) [inline src/sentry/sentry_metrics/querying/samples_list.py:312]: nit: should we move this to somewhere that can be shared?

🤔 also is there any casue where `self.query` could accidentally bleed into the other query terms? Should we wrap it in `()` just to be safe?

**Review by wmak — APPROVED** [review]
nit and a minor comment, otherwise lgtm

**Review by iambriccardo — APPROVED** [review]
LGTM

File: src/sentry/sentry_metrics/querying/samples_list.py (hunk 1 above)
Comment (Zylphrex) [reply src/sentry/sentry_metrics/querying/samples_list.py]: I've moved all the implicit conditions into the additional conditions so this is no longer an issue.

**Pull request merged**